controllers/appgroup_reconciler.go

```diff
@@ -36,25 +41,93 @@ data:
 	dummyConfigmapYAMLName = "templates/dummy-configmap.yaml"
 )
 
-func (r *ApplicationGroupReconciler) reconcile(ctx context.Context, l logr.Logger, appGroup *v1alpha1.ApplicationGroup) (bool, error) {
-	l = l.WithValues(v1alpha1.AppGroupNameKey, appGroup.Name)
-	l.V(3).Info("Reconciling ApplicationGroup object")
+func (r *ApplicationGroupReconciler) reconcileCreateOrUpdate(ctx context.Context, instance *v1alpha1.ApplicationGroup, patch client.Patch) error {
+	instance.Progressing()
```

Comment:
Might be a good idea to move the `r.Status().Patch(...)` into the `Progressing`/`RollingBack`/... methods since every update is followed by a patch (if it is conditional then we can have a flag `patch bool`.